src/core/lib/event_engine/grpc_polled_fd.h

```diff
@@ -46,8 +50,9 @@ class GrpcPolledFd {
   // Indicates if there is data left even after just being read from
   virtual bool IsFdStillReadableLocked() = 0;
   // Called once and only once. Must cause cancellation of any pending
-  // read/write callbacks.
-  virtual void ShutdownLocked(grpc_error_handle error) = 0;
+  // read/write callbacks. Return true when the Shutdown is confirmed, false
+  // otherwise.
+  virtual bool ShutdownLocked(absl::Status error) = 0;
```

Comment:
I think we need to add a couple of new comments to this API with this PR:

1) `error` must be of type: `Cancelled` to guarantee that this will do anything

2) Instead of calling this once and only once, we need to keep calling it until it returns `true`.

3) Should the return type by annotated with `GRPC_MUST_USE_RESULT` ?

Then we should have TODO that we could remove all of these requirements if we changed the `FdNode` lifetime model such that:
- FdNodes and their underlying socket handles remain alive for the lifetime of the resolver
- orphaning the resolver triggers shutdown and subsequent cleanup for all FdNodes and socket handles

Reply:
Done. Thanks!